File: WorldFeed/Client/src/app/core/Public/Parameters/CursorQueryParameters.ts
import {CustomRequestParameters, ICustomRequestParameters} from "./CustomRequestParameters";

// A query that can return multiple times based on a cursor value
export interface ICursorQueryParameters extends ICustomRequestParameters {
  // The cursor value to start the operation with
  cursor: string;

  // The maximum number of objects to return
  pageSize: number;
}

export class CursorQueryParameters extends CustomRequestParameters implements ICursorQueryParameters {
  constructor(parameters?: ICursorQueryParameters) {
    if (parameters) {
      super(parameters);

      this.cursor = parameters.cursor;
      this.pageSize = parameters.pageSize;
    } else {
      super();

      this.cursor = null;
      this.pageSize = 20;
    }
  }

  public cursor: string;

  public pageSize: number;
}

// public CursorQueryParameters()
// {
//   Cursor = null;
//   PageSize = 20;
// }
//
// public CursorQueryParameters(parameters: ICursorQueryParameters) : base(parameters)
// {
//   if (parameters == null) { return; }
//
//   Cursor = parameters.Cursor;
//   PageSize = parameters.PageSize;
// }

File: Sharebook/Server/src/Sharebook.Common/Public/Parameters/CursorQueryParameters.cs
namespace Sharebook.Common.Public.Parameters
{
    /// <summary>
    /// A query that can return multiple times based on a cursor value
    /// </summary>
    public interface ICursorQueryParameters : ICustomRequestParameters
    {
        /// <summary>
        /// Defines the cursor that will be used for executing the first request
        /// </summary>
        string Cursor { get; set; }

        /// <summary>
        /// Defines the maximum number of items that will be returned per request
        /// </summary>
        int PageSize { get; set; }
    }

    public class CursorQueryParameters : CustomRequestParameters, ICursorQueryParameters
    {
        public CursorQueryParameters()
        {
            Cursor = null;
            PageSize = 20;
        }

        public CursorQueryParameters(ICursorQueryParameters parameters) : base(parameters)
        {
            if (parameters == null)
            {
                return; 
            }

            this.Cursor = parameters.Cursor;
            this.PageSize = parameters.PageSize;
        }

        /// <inheritdoc/>
        public string Cursor { get; set; }

        /// <inheritdoc/>
        public int PageSize { get; set; }
    }
}
